oio/xcute/common/backend.py

```diff
@@ -471,58 +490,73 @@ def list_orchestrator_jobs(self, orchestrator_id):
         job_config_infos = pipeline.execute()
 
         return (
-            (job_id, self._unmarshal_job_conf(job_conf), self._unmarshal_job_info(job_info))
-            for job_conf, job_info in zip(*([iter(job_config_infos)] * 2))
+            (job_id, self._unmarshal_job_config(job_config),
+             self._unmarshal_job_info(job_info))
+            for job_config, job_info in zip(*([iter(job_config_infos)] * 2))
         )
 
     @handle_redis_exceptions
     def run_next(self, orchestrator_id):
         next_job = self.script_run_next(
-            keys=[orchestrator_id], client=self.conn)
+            keys=[self._get_timestamp(), orchestrator_id],
+            client=self.conn)
         if next_job is None:
             return None
         job_id, job_info, job_config = next_job
         job_info = self._unmarshal_job_info(
             self._lua_array_to_dict(job_info))
-        job_config = self._unmarshal_job_conf(
-            self._lua_array_to_dict(job_config))
-        return (job_id, job_info['job_type'], job_info.get('last_sent'),
-                job_config)
+        job_config = self._lua_array_to_dict(job_config)
+        return (job_id, job_info['job.type'], job_info.get('tasks.last_sent'),
+                self._unmarshal_job_config(job_config))
 
     @handle_redis_exceptions
     def free(self, job_id):
-        self.script_free(keys=[job_id], client=self.conn)
+        self.script_free(
+            keys=[self._get_timestamp(), job_id],
+            client=self.conn)
 
     @handle_redis_exceptions
     def fail(self, job_id):
-        self.script_fail(keys=[job_id], client=self.conn)
+        self.script_fail(
+            keys=[self._get_timestamp(), job_id],
+            client=self.conn)
 
     @handle_redis_exceptions
     def request_pause(self, job_id):
-        self.script_request_pause(keys=[job_id], client=self.conn)
+        self.script_request_pause(
+            keys=[self._get_timestamp(), job_id],
+            client=self.conn)
 
     @handle_redis_exceptions
     def resume(self, job_id):
-        self.script_resume(keys=[job_id], client=self.conn)
+        self.script_resume(
+            keys=[self._get_timestamp(), job_id],
+            client=self.conn)
 
     @handle_redis_exceptions
     def update_tasks_sent(self, job_id, task_ids, total_tasks,
                           all_tasks_sent=False):
         return self.script_update_tasks_sent(
-            keys=[job_id, total_tasks, str(all_tasks_sent)],
+            keys=[self._get_timestamp(), job_id, total_tasks,
+                  str(all_tasks_sent)],
             args=task_ids, client=self.conn)
 
     @handle_redis_exceptions
     def update_tasks_processed(self, job_id, task_ids,
                                task_errors, task_results):
         counters = dict()
         if task_errors:
```

Comment:
`task_errors` & `task_results` should never be empty. If we wanna double make sure they're not empty, get them with a `.get('foo', {})` in the orchestrator. That would remove two `ifs`